source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -647,24 +647,31 @@ Http::FilterDataStatus JsonTranscoderFilter::encodeData(Buffer::Instance& data,
   }
 
   response_in_.move(data);
-  if (encoderBufferLimitReached(response_in_.bytesStored())) {
+  if (encoderBufferLimitReached(response_in_.bytesStored() + response_out_.length())) {
     return Http::FilterDataStatus::StopIterationNoBuffer;
   }
 
   if (end_stream) {
     response_in_.finish();
   }
 
-  readToBuffer(*transcoder_->ResponseOutput(), data);
+  readToBuffer(*transcoder_->ResponseOutput(), response_out_);
```

Comment:
should we assert response_out_ is empty before readToBuffer()

same for line 699

Reply:
No, it does not need to be empty, `readToBuffer()` just appends data to the end.

It is not guaranteed to be empty either. If we get multiple `encodeData()` calls, only first call will be empty. Next calls will append data.